src/javascripts/ng-admin/Crud/delete/DeleteController.js

```diff
@@ -22,21 +23,29 @@ define(function () {
 
     DeleteController.prototype.deleteOne = function () {
         var notification = this.notification,
-            entityName = this.entity.name(),
-            $window = this.$window;
-
-        this.WriteQueries.deleteOne(this.view, this.entityId).then(function () {
-            this.back();
-            notification.log('Element successfully deleted.', { addnCls: 'humane-flatty-success' });
-        }.bind(this), function (response) {
-            // @TODO: share this method when splitting controllers
-            var body = response.data;
-            if (typeof body === 'object') {
-                body = JSON.stringify(body);
-            }
-
-            notification.log('Oops, an error occured : (code: ' + response.status + ') ' + body, {addnCls: 'humane-flatty-error'});
-        });
+            entityName = this.entity.name();
+
+        return this.WriteQueries.deleteOne(this.view, this.entityId)
+            .then(
+                function () {
+                    this.$state.go(this.$state.get('list'), angular.extend({
+                        entity: entityName
+                    }, this.$state.params));
+
+                    notification.log('Element successfully deleted.', { addnCls: 'humane-flatty-success' });
+                }.bind(this),
+                function (response) {
+                    // @TODO: share this method when splitting controllers
+                    var body = response.data;
+                    if (typeof body === 'object') {
+                        body = JSON.stringify(body);
+                    }
+
+                    this.back();
```

Comment:
Because of https://github.com/marmelab/ng-admin/pull/538

But true, there will be another issue here: what about deleting a record from a reference_many list?

A solution may be to add a "with_redirect" to true by default, and open the confirmation box in a pop-up. Reference many delete button would not redirect in this case. What do you think of it?